sway-core/src/asm_generation/from_ir.rs

```diff
@@ -1375,19 +1361,19 @@ impl<'ir> AsmBuilder<'ir> {
             });
         } else {
             let ret_reg = self.value_to_register(ret_val);
-            let size_in_bytes = ir_type_size_in_bytes(self.context, ret_type);
 
-            if size_in_bytes <= 8 {
+            if matches!(ret_type, Type::Unit | Type::Bool | Type::Uint(_)) {
```

Comment:
Is this the same as `is_copy_type`?

Reply:
Yeah, except different type -- AST type vs IR type. :)

Reply:
Ah true true. Probably doesn't hurt to add a similar method in the future for IR types.